thriftpy/contrib/aio/protocol.py

```diff
@@ -0,0 +1,441 @@
+# -*- coding: utf-8 -*-
```

Comment:
@wooparadog The asyncio support of #299 based on TFramedTransport transport which is easier to be implemented. This transport tries to read the whole message out from socket at the begin and save it into buffer, it is easier because it just needs yield once from read the whole message. My implementation based on another transport (TBinaryTransport) so it may be a little bit verbose.

Reply:
If it's unavoidable for simpler abstraction, there should be a better code path like:

* `contrib/aio/binary_transport`
* `contrib/aio/framed_transport`

but it hurts so much for all those duplicate code...